plugins/modules/zos_archive.py

```diff
@@ -0,0 +1,879 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2020, 2022
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: zos_archive
+version_added: "1.7.0"
+author:
+  - Oscar Fernando Flores Garcia (@fernandofloresg)
+short_description: Archive a dataset on z/OS.
+
+description:
+  - Creates or extends an archive.
+  - The source and archive are on the remote host,
+    and the archive is not copied to the local host.
+
+options:
+  path:
+    description:
+      - Remote absolute path, glob, or list of paths or globs for the file or files to compress or archive.
+    type: list
+    required: true
+    elements: str
+  format:
+    description:
+      - The type of compression to use.
+    type: dict
+    required: false
+    suboptions:
+      name:
+        description:
+          - The name of the format to use.
+        type: str
+        required: false
+        default: gz
+        choices:
+          - bz2
+          - gz
+          - tar
+          - zip
+          - terse
+          - xmit
+          - pax
+      format_options:
+        description:
+          - Options specific to each format.
+        type: dict
+        required: false
+        suboptions:
+          terse_pack:
+            description: Pack option to use for terse format.
+            type: str
+            choices:
+              - PACK
+              - SPACK
+          xmit_log_dataset:
+            description: Provide a name of data set to use for xmit log.
```

Comment:
1. I did not test this. Will add to beta.2 tests.

2. Added it into the docs.

3. XMIT creates the log data set automatically, if it doesn't exist already, in tests I don't test with an existing data set, but it will need to have those attributes in case it is already created.

4. I don't know this is a test item for beta.2.

Reply:
I think (1) I don't know if we need to test it, we call xmit once, xmit does seem to say its a 1:1 mapping of log file to xmit, if xmit does not like reuse i am sure it will propagate an error. 

(2) thank you

(3) Its an easy enough test, its more about if there is any error that will result , can our code recover, worth testing beta after beta.1.

(4) It might be worth creating a really small data set, like 4 CYL or something, its more about our code recovering than it is about testing XMIT.

I think 3 & 4 could be considered post beta.1 items to look into.

Reply:
The block here still had some issues so I typed it up for a copy/paste if you approve.
```
          xmit_log_data_set:
            description:
              - Provide the name of a data set to store xmit log output.
              - If the data set provided does not exist, the program will create it.
              - If the data set provided exists, the data set must have the following attributes:
                 LRECL=255, BLKSIZE=3120, and RECFM=VB
              - When providing the I(xmit_log_data_set) name, ensure there is adequate space.
```

Reply:
Done.

I have this tasks for beta 2.	
- What happens when XMIT log data set fills up? Propose to test filling the data set. Assign a small size into a provided one and try failing it.
	- Do we allow data set reuse? Do we allow data set reuse here? (seems like we can not based on the quoted text below, but its difficult to interpret "However, with any given invocation of the TRANSMIT or RECEIVE command, only one log entry is written to any one log data set."
	- MD5 use for checksums

Reply:
Nice